packages/excalidraw/renderer/renderElement.ts

```diff
@@ -457,6 +458,9 @@ const drawElementFromCanvas = (
   const zoom = elementWithCanvas.scale;
   let [x1, y1, x2, y2] = getElementAbsoluteCoords(element);
 
+  // need to use scene here since elementsMap is outdated when passed via args and breaks labelled arrows
+  const elementsMap = Scene.getScene(element)!.getNonDeletedElementsMap();
```

Comment:
`elementsMap` isn't outdated. It's just a subset that doesn't contain `appState.editingElement`.